workers/unity/Packages/com.improbable.gdk.core/Worker/WorkerBase.cs

```diff
@@ -1,108 +1,158 @@
+        public readonly ILogDispatcher LogDispatcher;
 
-            View = new MutableView(World, loggingDispatcher);
-            Origin = origin;
+        public readonly Type[] Systems = new[]
+        {
+            typeof(SpatialOSReceiveSystem),
+            typeof(SpatialOSSendSystem),
+            typeof(CleanReactiveComponentsSystem)
+        };
 
-            EntityGameObjectLinker = new EntityGameObjectLinker(World, View);
-        }
+        private readonly Dictionary<long, Entity> entityMapping;
 
-        public void Dispose()
+        protected WorkerBase(string workerId, ConnectionConfig config, Vector3 origin, EntityManager entityManager)
         {
-            WorkerRegistry.UnsetWorkerForWorld(this);
-            View.Dispose();
-            World.Dispose();
-        }
+            FindTranslationUnits();
+            // TODO addAllCommandRequestSenders(WorkerEntity, WorkerEntityId);
+            entityMapping = new Dictionary<long, Entity>();
 
-        public virtual void Connect(ConnectionConfig config)
-        {
+            LogDispatcher = new ForwardingDispatcher();
+            this.EntityManager = entityManager;
+            WorkerId = workerId;
             if (config is ReceptionistConfig)
             {
                 if (UseDynamicId)
                 {
-                    WorkerId = GenerateDynamicWorkerId();
+                    WorkerId = $"{GetWorkerType}-{Guid.NewGuid()}";
                 }
 
-                Connection = ConnectionUtility.ConnectToSpatial((ReceptionistConfig) config, GetWorkerType,
+                Connection = ConnectionUtility.ConnectToSpatial((ReceptionistConfig)config, GetWorkerType,
                     WorkerId);
             }
             else if (config is LocatorConfig)
             {
-                Connection = ConnectionUtility.LocatorConnectToSpatial((LocatorConfig) config, GetWorkerType);
+                Connection = ConnectionUtility.LocatorConnectToSpatial((LocatorConfig)config, GetWorkerType);
             }
             else
             {
                 throw new InvalidConfigurationException($"Invalid connection config was provided: '{config}' Only" +
                     "ReceptionistConfig and LocatorConfig are supported.");
             }
+            WorkerEntity = entityManager.CreateEntity(typeof(WorkerEntityTag));
 
             Application.quitting += () =>
             {
                 ConnectionUtility.Disconnect(Connection);
                 Connection = null;
             };
 
-            View.Connect();
+            entityManager.AddComponent(WorkerEntity, typeof(IsConnected));
+            entityManager.AddComponent(WorkerEntity, typeof(OnConnected));
+
+            Origin = origin;
+        }
+
+        public void Dispose()
+        {
+            ConnectionUtility.Disconnect(Connection);
+            EntityManager.DestroyEntity(WorkerEntity);
+            // todo destroy all spatial entities
```

Comment:
Do you mean GameObjects or ECS entities? As the world being disposed should handle cleaning all entities

Reply:
I don't think we want to dispose the world if we are disposing a worker though. The user might have other systems running which handle non-spatial entities.

Reply:
If they have non-spatial entities in a `SpatialOSWorld` instance that are not tied to the worker lifecycle - I'd argue that's incorrect

Reply:
discussed offline: there are use cases where we would the `SpatialOSWorld` to be persistent to e.g. handle reconnection logic and ensuring that a new worker gets spun up. If a user really wants to clean up non-spatial entities they would have to do one of the following 
1. create a new world and have them in there
2. dispose of this world and create a new one
3. have a separate system to clean them up properly